docs/developer-docs/smart-contracts/maintain/trapping.mdx

```diff
@@ -29,42 +35,31 @@ If a canister traps during a message execution, then two things happen:
 
+This is quite important as it might affect how one can implement canisters safely. Generally speaking, traps should be avoided in callback handler invocations. If this is not possible, then it's recommended to use the [cleanup system API call](/docs/current/references/ic-interface-spec#system-api-call) to ensure that any cleanup action is performed (e.g. releasing a lock that was acquired before making an inter-canister call).
 
 ## When to use traps explicitly
 
-
-Traps can also be useful in certain situations. If you encounter an unrecoverable error while processing a message, it might be better to trap and
-roll back any changes made in that message execution to avoid leaving your canister in a weird state. Another case would be if an invariant that
-should generally hold is violated -- again a trap might be the best option to ensure that there's no further damage done and the broken invariant
-can be investigated and fixed.
-
+Traps can also be useful in certain situations. If you encounter an unrecoverable error while processing a message, it might be better to trap and roll back any changes made in that message execution to avoid leaving your canister in a weird state. Another case would be if an invariant that should generally hold is violated.
 
 ## Troubleshooting why your canister trapped
 
 Some of the common root causes of canister traps include:
 
 - *“Heap out of bounds”*: Usually means that your canister is attempting to access heap memory that has not been allocated yet. This typically points to a programming error. Look for places where memory is allocated (e.g. creating vectors) and whether you try to access such memory before allocating it.
+
 - *“Stable memory out of bounds”*: Similar to *“heap out of bounds”* but for stable memory.
+
 - *“Integer division by zero”*: The canister attempted to divide by zero. Inspect the canister's code for any division operations.
-- *“Unreachable”*: Typically produced when the canister panic's, e.g. for Rust canisters. Unfortunately, the system does not capture a backtrace at the moment, so you won't see more information. It's recommended to use the [panic hook](https://github.com/dfinity/cdk-rs/blob/08cbe397fcbaf62dd1edbec218492f967e80dd00/src/ic-cdk/src/printer.rs#L5) so that panic's are converted to an explicit call to [ic0.trap](/docs/current/references/ic-interface-spec#debugging-aids) which will append an error message that will be more useful for debugging.
+
+- *“Unreachable”*: Typically produced when the canister panic's, e.g. for Rust canisters. It's recommended to use the [panic hook](https://github.com/dfinity/cdk-rs/blob/08cbe397fcbaf62dd1edbec218492f967e80dd00/src/ic-cdk/src/printer.rs#L5) so that panic's are converted to an explicit call to [ic0.trap](/docs/current/references/ic-interface-spec#debugging-aids) which will append an error message that will be more useful for debugging.
```

Comment:
```suggestion
- *“Unreachable”*: Typically produced when the canister panics, e.g. for Rust canisters. It's recommended to use the [panic hook](https://github.com/dfinity/cdk-rs/blob/08cbe397fcbaf62dd1edbec218492f967e80dd00/src/ic-cdk/src/printer.rs#L5) so that panics are converted to an explicit call to [ic0.trap](/docs/current/references/ic-interface-spec#debugging-aids) which will append an error message that will be more useful for debugging.
```